api/src/main/java/marquez/db/StatsDao.java

```diff
@@ -5,68 +5,104 @@
+                                      AND event_time < DATE_TRUNC('hour', now()) + INTERVAL '1 hour') current_hour
+                                   ON current_hour.start_interval = hs.start_interval
+                ORDER BY hs.start_interval;
+              """)
   List<LineageMetric> getLastDayMetrics();
 
   @SqlQuery(
       """
-          WITH day_series AS (
-              SELECT
-                  generate_series(
-                          DATE_TRUNC('day', NOW() - INTERVAL '7 days'),  -- Start 7 days ago
-                          DATE_TRUNC('day', NOW()) - INTERVAL '1 day',   -- End at the start of today
-                          '1 day'
-                  ) AS start_interval
-          )
-          SELECT
-              ds.start_interval,
-              ds.start_interval + INTERVAL '1 day' AS end_interval,
-              COALESCE(mv.fail, 0) AS fail,
-              COALESCE(mv.start, 0) AS start,
-              COALESCE(mv.complete, 0) AS complete,
-              COALESCE(mv.abort, 0) AS abort
-          FROM
-              day_series ds
-                  LEFT JOIN
-              lineage_events_by_type_daily_view mv
-              ON
-                  ds.start_interval = mv.start_interval
-          ORDER BY
-              ds.start_interval;
-          """)
-  List<LineageMetric> getLastWeekMetrics();
+        WITH local_now AS (
+            SELECT (NOW() AT TIME ZONE :timezone) AS local_now
+        ),
+             day_series AS (
+                 SELECT generate_series(
+                                DATE_TRUNC('day', ln.local_now - INTERVAL '6 days'),
+                                DATE_TRUNC('day', ln.local_now),
+                                '1 day'
+                        ) AS start_interval
+                 FROM local_now ln
+             )
+        SELECT
+            ds.start_interval,
+            ds.start_interval + INTERVAL '1 day' AS end_interval,
+            COALESCE(current_day.fail, mv.fail, 0) AS fail,
+            COALESCE(current_day.start, mv.start, 0) AS start,
+            COALESCE(current_day.complete, mv.complete, 0) AS complete,
+            COALESCE(current_day.abort, mv.abort, 0) AS abort
+        FROM day_series ds
+                 LEFT JOIN lineage_events_by_type_daily_view mv
+                           ON ds.start_interval = (mv.start_interval AT TIME ZONE 'UTC') AT TIME ZONE :timezone
+                 LEFT JOIN (
+            SELECT
+                DATE_TRUNC('day', ln.local_now) AS start_interval,
+                DATE_TRUNC('day', ln.local_now) + INTERVAL '1 day' AS end_interval,
+                COALESCE(SUM(CASE WHEN state = 'FAIL' THEN 1 ELSE 0 END), 0) AS fail,
+                COALESCE(SUM(CASE WHEN state = 'START' THEN 1 ELSE 0 END), 0) AS start,
+                COALESCE(SUM(CASE WHEN state = 'COMPLETE' THEN 1 ELSE 0 END), 0) AS complete,
+                COALESCE(SUM(CASE WHEN state = 'ABORT' THEN 1 ELSE 0 END), 0) AS abort
+            FROM current_day_lineage_metrics
```

Comment:
Minor: Do you need the cross join? You can reference the CTE directly:

```
FROM current_day_lineage_metrics, local_now
```

Reply:
Yeah, that is simpler for sure... Good point.